File: jetpack/lib-arch/src/main/java/com/yyxnb/oh/arch/lifecycle/IActivityLifecycle.java
package com.yyxnb.oh.arch.lifecycle;

import android.app.Activity;
import android.os.Bundle;

import com.yyxnb.oh.core.interfaces.ILifecycle;

/**
 * ================================================
 * 作    者：yyx
 * 版    本：1.0
 * 日    期：2020/11/21
 * 历    史：
 * 描    述：IActivityDelegate
 * ================================================
 */
public interface IActivityLifecycle extends ILifecycle {

    void onCreate(Bundle savedInstanceState);

    void onStart();

    void onResume();

    void onPause();

    void onStop();

    void onSaveInstanceState(Activity activity, Bundle outState);

    void onDestroy();

}

File: lib_arch/src/main/java/com/yyxnb/arch/interfaces/IActivityLifecycle.kt
package com.yyxnb.arch.interfaces

import android.app.Activity
import android.os.Bundle

/**
 * activity 代理接口
 */
interface IActivityLifecycle {

    fun onCreate(savedInstanceState: Bundle?)

    fun onStart()

    fun onResume()

    fun onPause()

    fun onStop()

    fun onSaveInstanceState(activity: Activity?, outState: Bundle?)

    fun onDestroy()
}